website/src/models/group.js

```diff
@@ -404,11 +404,29 @@ GroupSchema.methods.leave = function(user, keep, mainCb){
           group.type === 'party' ||
           (group.type === 'guild' && group.privacy === 'private')
       )){
-        // TODO remove invitations to this group
-        groupWasRemoved = true;
-        Group.remove({
-          _id: group._id
-        }, cb);
+        async.waterfall([
```

Comment:
This should probably be handled in a pre or post hook for remove: http://mongoosejs.com/docs/middleware.html